flink-streaming-java/src/test/java/org/apache/flink/streaming/api/DataStreamTest.java

```diff
@@ -906,6 +919,256 @@ public void testChannelSelectors() {
 	}
 
 	/////////////////////////////////////////////////////////////
+	// KeyBy testing
+	/////////////////////////////////////////////////////////////
+
+	@Rule
+	public ExpectedException expectedException = ExpectedException.none();
```

Comment:
First time I'm seeing this pattern and i can't help but feel that a simple try-catch block would be more readable.